### ISSUE TITLE
externalize your parallel programming library

### ISSUE THREAD

**Issue by UnixJunkie** [facebook/pyre-check#77]
Hello,

Someone told me that there is a shared memory parallel programming library
hidden in there, under hack_parallel/.

It would be quite nice if this could be released as a separate library,
so that the whole ocaml community could benefit from it.

Thanks a lot,
F.

**Comment by UnixJunkie** [comment]
@mbouaziz I even wonder if other facebook ocaml projects could not benefit from it

**Comment by dark** [comment]
Hi @UnixJunkie, we are indeed using a shared memory library that we borrowed from the [hhvm](https://github.com/facebook/hhvm) project. The authoritative copy lives in that repository at https://github.com/facebook/hhvm/tree/master/hphp/hack/src.

We have adapted this library to our needs by removing files we don't need, and are lagging behind in terms of patches and bugfixes. Therefore, you might have a better chance by pinging the maintainers of that repository with this suggestion.

Last but not least, the flow project @ FB uses this library as well: https://github.com/facebook/flow/tree/master/hack :)
